### ISSUE TITLE
Stage 1 pre-training

### ISSUE THREAD

**Issue by ChantalMP** [salesforce/LAVIS#276]
Hi,
I'm trying to train BLIP-2 on a different image domain, so I'm also using a different image encoder. I assume it will be important to perform stage 1 pre-training to learn to extract meaningful visual features with this encoder from the new images. Would you agree? And do you think it could still be meaningful to utilize the already pre-trained Q-Former weights?

Along these lines it would of course be amazing if the stage 1 pre-training code would be published in some way. Are there any plans towards this?

Best regards and thanks for your great work!

**Comment by LiJunnan1992** [comment]
Yes it is suggested to run stage 1 pre-training from scratch for a new visual encoder. You could run BLIP-2 stage 1 pre-training now with `bash run_scripts/blip2/train/pretrain_stage1.sh`. If you wish to load BLIP-2 stage-1 checkpoint, just set load_pretrained to True in [pretrain_stage1.yaml](https://github.com/salesforce/LAVIS/blob/main/lavis/projects/blip2/train/pretrain_stage1.yaml).

**Comment by ChantalMP** [comment]
Thanks a lot for your answer and for publishing the pretraining code! 😊
